Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -412,7 +391,7 @@ public Task<IUploadProgress> UploadAsync()
 
             try
             {
-                UploadUri = await InitializeUpload(cancellationToken).ConfigureAwait(false);
+                UploadUri = await InitiateSessionAsync(cancellationToken).ConfigureAwait(false);

```

Comment:
Right. Okay, that's not *too* bad. Let's still keep the "here's the URI we're going to tell the caller about in the virtual method" separate from "here's the URI populated at the start of UploadAsync".